app/services/crash-reporter.ts

```diff
@@ -94,7 +95,14 @@ export class CrashReporterService extends Service {
   }
 
   private writeStateFile(code: EAppState) {
-    this.appState = { code, version: this.version };
+    this.appState = { code, version: this.version, detected: '' };
+    this.appState = this.readStateFile();
+    this.appState.code = code;
+    this.appState.version = this.version;
+    if (this.appState.code !== EAppState.Starting) {
+      this.appState.detected = '';
+    }
```

Comment:
this could be simplified to
```
this.appState = {...this.readStateFile, code};
if (this.appState.code !== EAppState.Starting) {
      this.appState.detected = '';
}
```

Reply:
It does not let me to use ... in this way. 

Removed unnecessary preinitialization.